scipy/sparse/base.py

```diff
@@ -537,8 +557,32 @@ def __getattr__(self, attr):
         else:
             raise AttributeError(attr + " not found")
 
-    def transpose(self):
-        return self.tocsr().transpose()
+    def transpose(self, axes=None, copy=False):
+        """
+        Reverses the dimensions of the sparse matrix.
+
+        Parameters
+        ----------
+        axes : None, optional
+            This argument is in the signature *solely* for NumPy
+            compatibility reasons, as reversing the dimensions is
+            the only logical permutation that should be requested.
```

Comment:
The issue that I have is that is the additional flexibility worth the extra validation complexity?  There are AFAICT _thirteen_ "valid" tuples that you can pass in based on the `np.matrix` functionality, and if we're going to support tuples, I would think that all of them need to be accepted.  It seems simpler IMHO to not support them since none of them are actually any more meaningful than the parameters I am currently accepting.